types/actions/omelasticsearch.pp

```diff
@@ -0,0 +1,32 @@
+type Rsyslog::Actions::Omelasticsearch = Struct[{
+  Server               => Optional[Variant[String[1], Array[String[1]]]],
+  Serverport           => Optional[Integer],
+  HealthCheckTimeout   => Optional[Integer],
+  searchIndex          => Optional[String[1]],
+  dynSearchIndex       => Optional[Enum['on', 'off']],
+  searchType           => Optional[String[1]],
+  dynSearchType        => Optional[Enum['on', 'off']],
+  pipelineName         => Optional[String[1]],
+  dynPipelineName      => Optional[Enum['on', 'off']],
+  asyncrepl            => Optional[Enum['on', 'off']],
+  usehttps             => Optional[Enum['on', 'off']],
+  timeout              => Optional[Pattern[/^([0-9]+)(ms|s|m)$/]],
```

Comment:
I think this would be the appropriate pattern for timeout in `ms`, `s`, or `m` for `milliseconds`, `seconds`, and `minutes` respectively. Can someone review this to ensure it's correct?

Reply:
I tested manually, it looks to do the job. But it should be interesting to have it in tests.